src/input/input_handler.cpp

```diff
@@ -0,0 +1,694 @@
+    case OrbisPadButtonDataOffset::Left:
+        return SDL_GAMEPAD_BUTTON_DPAD_LEFT;
+    case OrbisPadButtonDataOffset::Right:
+        return SDL_GAMEPAD_BUTTON_DPAD_RIGHT;
+    case OrbisPadButtonDataOffset::Options:
+        return SDL_GAMEPAD_BUTTON_START;
+    case (OrbisPadButtonDataOffset)BACK_BUTTON:
+        return SDL_GAMEPAD_BUTTON_BACK;
+
+    default:
+        return ((u32)-1) - 0x10000000;
+    }
+}
+u32 GetControllerButtonInputId(u32 cbutton) {
+    if ((cbutton & ((u32)OrbisPadButtonDataOffset::TouchPad | LEFTJOYSTICK_HALFMODE |
+                    RIGHTJOYSTICK_HALFMODE)) != 0) {
+        return (u32)-1;
+    }
+    return GetOrbisToSdlButtonKeycode((OrbisPadButtonDataOffset)cbutton) + 0x10000000;
+}
+
+// syntax: 'name, name,name' or 'name,name' or 'name'
+InputBinding GetBindingFromString(std::string& line) {
+    u32 key1 = 0, key2 = 0, key3 = 0;
+
+    // Split the string by commas
+    std::vector<std::string> tokens;
+    std::stringstream ss(line);
+    std::string token;
+
+    while (std::getline(ss, token, ',')) {
+        tokens.push_back(token);
+    }
+
+    // Check and process tokens
+    for (const auto& t : tokens) {
+        if (string_to_keyboard_key_map.find(t) != string_to_keyboard_key_map.end()) {
+            // Map to keyboard key
+            u32 key_id = string_to_keyboard_key_map.at(t);
+            if (!key1)
+                key1 = key_id;
+            else if (!key2)
+                key2 = key_id;
+            else if (!key3)
+                key3 = key_id;
+        } else if (string_to_axis_map.find(t) != string_to_axis_map.end()) {
+            // Map to axis input ID
+            u32 axis_id = GetAxisInputId(string_to_axis_map.at(t));
+            if (axis_id == (u32)-1) {
+                return InputBinding(0, 0, 0);
+            }
+            if (!key1)
+                key1 = axis_id;
+            else if (!key2)
+                key2 = axis_id;
+            else if (!key3)
+                key3 = axis_id;
+        } else if (string_to_cbutton_map.find(t) != string_to_cbutton_map.end()) {
+            // Map to controller button input ID
+            u32 cbutton_id = GetControllerButtonInputId((u32)string_to_cbutton_map.at(t));
+            if (cbutton_id == (u32)-1) {
+                return InputBinding(0, 0, 0);
+            }
+            if (!key1)
+                key1 = cbutton_id;
+            else if (!key2)
+                key2 = cbutton_id;
+            else if (!key3)
+                key3 = cbutton_id;
+        } else {
+            // Invalid token found; return default binding
+            return InputBinding(0, 0, 0);
+        }
+    }
+
+    return InputBinding(key1, key2, key3);
+}
+
+// function that takes a controlleroutput, and returns the array's corresponding element's pointer
+ControllerOutput* GetOutputPointer(const ControllerOutput& parsed) {
+    // i wonder how long until someone notices this or I get rid of it
+    int i = 0;
```

Comment:
Avoid writing "smart-code" if possible. I dont see any apparent benefit to this and will confuse readers.
I also wonder if this function is necessary, since std::ranges::find() can do the same